crates/noirc_evaluator/src/ssa_refactor/ir/dfg.rs

```diff
@@ -267,6 +243,8 @@ impl DataFlowGraph {
         parameter
     }
 
+    /// Insert an instruction at the end of a given block.
+    /// If the block already has a terminator, the instruction is inserted before the terminator.
```

Comment:
Why is it useful to insert an instruction after a terminator has already been added?

Reply:
It is not necessarily useful, but since this would be the behavior if this function was called after a terminator was set, I thought it better to document it than leave it unspecified. The reasoning for this is we store instructions separately from the terminator `{ instructions: Vec<InstructionId>, terminator: Option<TerminatorInstruction> }` so if we're just pushing to the instructions Vec it is naturall the terminator stays at the end rather than somehow winding up in the middle of the Vec - which the type system wouldn't allow in the first place.